### DISCUSSION TITLE
About the reason to delete the safe browsing code

### DISCUSSION THREAD

**Discussion by uazo** [bromite/bromite#1528] [Q&A]
I wanted to answer your question if it is possible to use safe browsing only with local db and so I looked at the code.

Yes, it is possible. but

1) it is not possible to redistribute the db, at least, in their license it is not expressly prohibited but there is no one who has made it freely available
2) therefore, an api key is needed to enable the download of the hash list. the key is free but subject to some limits (10,000 clients per 24-hour period) and, as far as I know, a credit card must be provided for activation (but this should be checked).
3) there is a limit imposed by the license it says

> "You may not treat a URL from Google's list as an unsafe web resource, such as by showing users a warning about the site or blocking access to it, unless your application has received from Google updated information (via the applicable API method) within the **past thirty minutes**. "

ref. https://developers.google.com/safe-browsing/terms

although in the chromium code I have not yet found where that is enforced.
thus, we would be forced to ping the google server every 30 minutes.

4) the key is present without any protection in the apk, so much so that it is possible to extract the api used by ~~chromium vanilla~~ chrome by (easily) decompiling the apk.

moreover, looking at the safebrowsing code, it seems to me that all the flags that allow the deactivation of the functionality from the settings already exist, which would allow us to eliminate that extremely invasive patch (like the logic applied to floc code), unless there are others reasons, that I do not know.

at the most it is possible to force the deactivation of requests to google servers, inhibiting the code that makes the call and inserting some mechanism that warns us of new calls, perhaps directly at the time of applying the patch.

**Comment by csagan5** [comment]
> I wanted to answer your question if it is possible to use safe browsing only with local db and so I looked at the code.

This is something that also @thestinger looked into; back then and until now I was not aware of the content of such local db. Considering that it is proprietary data, I have no interest in shipping it with Bromite.



> moreover, looking at the safebrowsing code, it seems to me that all the flags that allow the deactivation of the functionality from the settings already exist, which would allow us to eliminate that extremely invasive patch (like the logic applied to floc code), unless there are others reasons, that I do not know.
> 
> at the most it is possible to force the deactivation of requests to google servers, inhibiting the code that makes the call and inserting some mechanism that warns us of new calls, perhaps directly at the time of applying the patch.

The invasive patch is there mostly because it is quite difficult to figure out what the code is doing, and it is often in high flux. If you can figure out a less invasive patch that is also as efficient, we can use it. As for the local db: see my previous comment.

**Reply by uazo** [reply]
>  If you can figure out a less invasive patch that is also as efficient

based on what I am doing #1569 (which always started from #1516), I am considering another hypothesis.
the patch as it is must be kept, but perhaps only to verify that there is no new code to modify.
let me explain: it could be useful to have two distinct patches per area, one only for verification, and another to apply to bromite, where the first should be applied to chromium without any patch.
maybe this is double work, I don't know yet.

**Reply by csagan5** [reply]
> apply to bromite, where the first should be applied to chromium without any patch.
> maybe this is double work, I don't know yet.

The Chromium build should be as close as possible to the upstream Chromium build, otherwise we cannot test bugs with it.
